### PR TITLE
Backport crash fixes to v3.7.x

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

Backports #27755 for the macOS crash, #27785 for PGF on Windows, and #26689 just to make the previous backport easier.

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [n/a] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [n/a] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
Test failures are because #27624 was not backported but pytest v8 is being used.

We can either add that to the backports or pin pytest on this branch.

**Review by tacaswell — APPROVED** [review]

**Comment by QuLogic** [comment]
Hmm, it seems like Python 3.8 and Pytest 8's new context isn't working well together for some reason.

**Comment by tacaswell** [comment]
Maybe we should pin pytest back to less than 8 for this branch instead?  It would be less code changes.

**Comment by ksunden** [comment]
https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers

I'm confused as to why it works on 3.9, though... using parens in a `with` statement was apparently new in 3.10

File: lib/matplotlib/tests/test_ticker.py [hunk 1]
```diff
@@ -730,10 +731,17 @@ def test_mathtext_ticks(self):
...
-            ax.set_xticks([-1, 0, 1])
-            fig.canvas.draw()
+        if parse_version(pytest.__version__).major < 8:
+            with pytest.warns(UserWarning, match='cmr10 font should ideally'):
+                fig, ax = plt.subplots()
+                ax.set_xticks([-1, 0, 1])
+                fig.canvas.draw()
+        else:
+            with (pytest.warns(UserWarning, match="Glyph 8722"),
+                  pytest.warns(UserWarning, match='cmr10 font should ideally')):
```
Comment (ksunden) [inline lib/matplotlib/tests/test_ticker.py:741]:
```suggestion
            with pytest.warns(UserWarning, match="Glyph 8722"), \
                  pytest.warns(UserWarning, match='cmr10 font should ideally'):
```

**Review by dstansby — APPROVED** [review]

**Pull request merged**